### PR TITLE
Upgrade parent-pom to latest version and use its dependency-check

### PR DESCRIPTION
**Pull request by bcullerton**
Move the dependency-check configuration to parent-pom and remove from application level. Also incorporates new api key parameter which allows downloads from NIST api. NIST are retiring data feeds so the key and therefore dep check up version will become necessary: https://github.com/companieshouse/parent-pom/blob/master/pom.xml#L85

### Work checklist

- [ ] Tests added where applicable

### Merge instructions

We are committed to keeping commit history clean, consistent and linear. To achieve this, this commit should be structured as follows:

```
<type>[optional scope]: <description>
```

and contain the following structural elements:

- fix: a commit that patches a bug in your codebase (this correlates with PATCH in semantic versioning),
- feat: a commit that introduces a new feature to the codebase (this correlates with MINOR in semantic versioning),
- BREAKING CHANGE: a commit that has a footer `BREAKING CHANGE:` introduces a breaking API change (correlating with MAJOR in semantic versioning). A BREAKING CHANGE can be part of commits of any type,
- types other than `fix:` and `feat:` are allowed, for example `build:`, `chore:`, `ci:`, `docs:`, `style:`, `refactor:`, `perf:`, `test:`, and others,
- footers other than `BREAKING CHANGE: <description>` may be provided.

### CODE REVIEW

**Review by mwestacott — APPROVED** [review]
LGTM.

Suggest awaiting one more approval from @dlloyd1-ch or @markpit before merging.

**Review by markpit — APPROVED** [review]

**Pull request merged**
